sanitycheck: key error issue while trying to access the key in dict

yerabolu opened the pull request:
While generating json report, status key does not exist and throws
keyerror in some specific scenarios.get() method is used to avoid
error.

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented on scripts/sanity_chk/sanitylib.py:
```diff
@@ -3502,7 +3502,7 @@ def json_report(self, filename, platform=None, append=False, version="NA"):
                     test_details = i["test_details"]
                     for test_data in test_details:
-                        if not (test_data["status"]) == "failed":
+                        if not (test_data.get("status")) == "failed":
```
this conditional is awkward, why not 

```
if test.data.get(...) != "failed"
```
or whatever you are trying to check for here. It will still fail otherwise I think.

nashif commented:
FYI: assignee is the maintainer, not the contributor :)

yerabolu committed:
sanitycheck: key error issue while trying to access the key in dict

While generating json report, status key does not exist and throws
keyerror in some specific scenarios.get() method is used to avoid
error.

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented:
this PR needs to be resubmitted from a  fork.
